vector/src/main/java/org/apache/arrow/vector/Decimal256Vector.java

```diff
@@ -566,9 +566,7 @@ private class TransferImpl implements TransferPair {
     Decimal256Vector to;
 
     public TransferImpl(String ref, BufferAllocator allocator) {
-      to =
-          new Decimal256Vector(
-              ref, allocator, Decimal256Vector.this.precision, Decimal256Vector.this.scale);
+      to = new Decimal256Vector(ref, Decimal256Vector.this.field.getFieldType(), allocator);
```

Comment:
Should we not test with `field`/`getFiedlType()` is not null and fallback to the vector precision if null ?